File: src/test/java/com/github/xumeng/java/problems/easy/problem0290/Solution0290Test.java
package com.github.xumeng.java.problems.easy.problem0290;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0290Test {

    private final Solution0290 solution = new Solution0290();

    @Test
    public void test1() {
        String pattern = "abba";
        String s = "dog cat cat dog";
        boolean suitable = solution.wordPattern(pattern, s);
        assertTrue(suitable);
    }

    @Test
    public void test2() {
        String pattern = "abba";
        String s = "dog cat cat fish";
        boolean suitable = solution.wordPattern(pattern, s);
        assertFalse(suitable);
    }

    @Test
    public void test3() {
        String pattern = "aaaa";
        String s = "dog cat cat dog";
        boolean suitable = solution.wordPattern(pattern, s);
        assertFalse(suitable);
    }

    @Test
    public void test4() {
        String pattern = "a";
        String s = "dog";
        boolean suitable = solution.wordPattern(pattern, s);
        assertTrue(suitable);
    }

    @Test
    public void test5() {
        String pattern = "a";
        String s = "dog cat";
        boolean suitable = solution.wordPattern(pattern, s);
        assertFalse(suitable);
    }

    @Test
    public void test6() {
        String pattern = "ab";
        String s = "cat cat";
        boolean suitable = solution.wordPattern(pattern, s);
        assertFalse(suitable);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0290/Solution0290Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0290

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0290Test {

    private val solution = Solution0290()

    @Test
    fun test1() {
        val pattern = "abba"
        val s = "dog cat cat dog"
        val suitable = solution.wordPattern(pattern, s)
        assertTrue(suitable)
    }

    @Test
    fun test2() {
        val pattern = "abba"
        val s = "dog cat cat fish"
        val suitable = solution.wordPattern(pattern, s)
        assertFalse(suitable)
    }

    @Test
    fun test3() {
        val pattern = "aaaa"
        val s = "dog cat cat dog"
        val suitable = solution.wordPattern(pattern, s)
        assertFalse(suitable)
    }

    @Test
    fun test4() {
        val pattern = "a"
        val s = "dog"
        val suitable = solution.wordPattern(pattern, s)
        assertTrue(suitable)
    }

    @Test
    fun test5() {
        val pattern = "a"
        val s = "dog cat"
        val suitable = solution.wordPattern(pattern, s)
        assertFalse(suitable)
    }

    @Test
    fun test6() {
        val pattern = "ab"
        val s = "cat cat"
        val suitable = solution.wordPattern(pattern, s)
        assertFalse(suitable)
    }

}